components/CreateEventsWizard.tsx

```diff
@@ -0,0 +1,353 @@
+  useEffect(() => {
+    setValid(true);
+    if (!name || !teachers) {
+      setValid(false);
+    }
+    if (!minLevel || (multipleLevels && (!maxLevel || minLevel >= maxLevel))) {
+      setValid(false);
+    }
+    if (!startDate || !startTime || !endTime) {
+      setValid(false);
+    }
+  }, [name, multipleLevels, minLevel, maxLevel, startDate, startTime, endTime, color, teachers]);
+
+  // force max level to exceed min level
+  useEffect(() => {
+    if (minLevel >= maxLevel) {
+      setMaxLevel(minLevel + 1);
+    }
+  }, [minLevel, maxLevel]);
+
+  // callback for hiding modal on close
+  const handleRepeatClose = (): void => {
+    setShowRepeatModal(false);
+  };
+
+  // callback for setting repeat modal states on save
+  const handleRepeatStates = (
+    repeat_param: boolean,
+    interval_param: number,
+    weekDays_param: number[],
+    endType_param: string,
+    endDate_param: Date,
+    count_param: number
+  ): void => {
+    setRepeat(repeat_param);
+    if (repeat_param) {
+      setInterval(interval_param);
+      setWeekDays(weekDays_param);
+      setEndType(endType_param);
+      setEndDate(endDate_param);
+      setCount(count_param);
+
+      // rrule text to display on button
+      const rule = new RRule({
+        interval: interval_param,
+        freq: RRule.WEEKLY,
+        byweekday: weekDays_param,
+      });
+      setRruleText(rule.toText());
+    } else {
+      setRruleText("Never Repeat");
+    }
+  };
+
+  // handles create wizard submit
+  const handleSubmit = async (): Promise<void> => {
+    setErrMsg("");
+    setLoading(true);
+    let lang = "unknown";
+    if (name.toLowerCase().includes("java")) {
+      lang = "Java";
+    } else if (name.toLowerCase().includes("python")) {
+      lang = "Python";
+    }
+
+    // construct rrule object
+    let rrule;
+    if (repeat) {
+      if (endType === "on") {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+          until: endDate,
+        });
+      } else if (endType === "after") {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+          count: count,
+        });
+      } else {
+        rrule = new RRule({
+          dtstart: startDate,
+          interval: interval,
+          freq: RRule.WEEKLY,
+          byweekday: weekDays,
+        });
+      }
+    } else {
+      rrule = new RRule({
+        dtstart: startDate,
+        count: 1,
+      });
+    }
+    const rruleStr = rrule.toString();
+
+    const createEvent: CreateClassEvent = {
+      name: name,
+      startTime: startTime,
+      endTime: endTime,
+      timeZone: DateTime.local().zoneName,
+      rrule: rruleStr,
+      language: lang,
+      neverEnding: repeat && endType === "never",
+      backgroundColor: colorMap[color],
+      teachers: teachers.split(","),
```

Comment:
can we add a trim to each element inside this? Just so that spaces don't get in the way